### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by belmeetmule**
In this PR, I and my coding partner have done the following activities:

- [ ] solved the N+1 problem caused when fetching all posts and their comments for a user
**Before the N+1 problem is solved !**
n-plus-one-before
**After the N+1 problem is solved !**
n-plus-one-after

- [ ] Integration specs

-  **Configured Capybara**
- **Wrote integration tests for each view of the project**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,255 +1,259 @@
-<a name="readme-top"></a>
```
Comment (rustam-tolipov) [inline README.md:1]: **You did a great job!**
However, we have some cleaning jobs to do. 

- [ ] Kindly remove commented code from your `index.html.erb_spec.rb`:


Screenshot 2023-02-07 at 20 48 55

File: README.md [hunk 2]
```diff
@@ -1,255 +1,259 @@
...
-
```
Comment (rustam-tolipov) [inline README.md:2]: - [ ] In requests/posts_spec.rb
Kindly remove unused commented codes!

Screenshot 2023-02-07 at 20 49 22

File: README.md [hunk 3]
```diff
@@ -1,255 +1,259 @@
...
-<!--
-HOW TO USE:
```
Comment (rustam-tolipov) [inline README.md:4]: - [ ] In request/comments_spec.rb
Kindly remove unused commented codes!

Screenshot 2023-02-07 at 20 49 30

File: README.md [hunk 4]
```diff
@@ -1,255 +1,259 @@
...
-This is an example of how you may give instructions on setting up your project locally.
-
```
Comment (rustam-tolipov) [inline README.md:6]: - [ ] When we generate modes with scaffold it creates some unnecessary helpers. At that point, I think these files are not necessary for us to keep. Kindly remove them as well.

Screenshot 2023-02-07 at 20 49 46

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @belmeetmule,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @belmeetmule 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Specs passed
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — COMMENTED** [review]
Hi Mulugeta, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure that your ReadMe is professionally documented:
- Kindly ensure that provided placeholder texts/ guide instructions are removed and/or replaced with appropriate contents applicable under each section of your ReadMe. 

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**